pallets/stateful-storage/src/tests/child_tree_tests.rs

```diff
@@ -0,0 +1,132 @@
+use crate::{
+	stateful_child_tree::StatefulChildTree, test_common::test_constants::*, tests::mock::*,
+};
+use common_primitives::schema::{ModelType, PayloadLocation, SchemaId};
+use frame_support::BoundedVec;
+#[allow(unused_imports)]
+use pretty_assertions::{assert_eq, assert_ne, assert_str_eq};
+use sp_core::ConstU32;
+use sp_std::collections::btree_set::BTreeSet;
+
+#[test]
+fn child_tree_write_read() {
+	new_test_ext().execute_with(|| {
+		// arrange
+		let pallet_name: &[u8] = b"test-pallet";
+		let storage_name_1: &[u8] = b"storage1";
+		let msa_id = 1;
+		let schema_id: SchemaId = 2;
+		let page_id: u8 = 3;
+		let keys = &(schema_id, page_id);
+		let val = TestStruct {
+			model_type: ModelType::AvroBinary,
+			payload_location: PayloadLocation::OnChain,
+			number: 8276387272,
+		};
+
+		// act
+		<StatefulChildTree>::write(&msa_id, pallet_name, storage_name_1, keys, &val);
+
+		// assert
+		let read =
+			<StatefulChildTree>::try_read(&msa_id, pallet_name, storage_name_1, keys).unwrap();
+		assert_eq!(Some(val), read);
+	});
+}
+
+type TestKeyString = BoundedVec<u8, ConstU32<16>>;
+type TestKey = (TestKeyString, TestKeyString, u64);
+
+#[test]
+fn child_tree_iterator() {
```

Comment:
this should be in child tree file

Reply:
hmm.  Isn't it in `child_tree_tests.rs` already?

Reply:
oops for some reason I thought it was in apply_item_action. my bad.